src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ClassDefHandler.java

```diff
@@ -79,6 +79,10 @@ public void checkIndentation() {
                     logError(ident, "ident", lineStart);
                 }
             }
+            if (getMainAst().getType() == TokenTypes.RECORD_DEF) {
+                checkRightParen(getRecordComponentsLeftParen(),
+                    getRecordComponentsRightParen());
+            }
```

Comment:
Left parenthesis is already checked by LineWrappingHandler; similar duplication in logic is likely to be why we have unresolvable indentation issue at https://github.com/checkstyle/checkstyle/issues/9269